tests/integration/kubernetes/k8s-guest-pull-image-authenticated.bats

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bats
+# Copyright (c) 2024 IBM Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/confidential_common.sh"
+load "${BATS_TEST_DIRNAME}/confidential_kbs.sh"
+
+export KBS="${KBS:-false}"
+
+setup() {
+    # Log checking not working on TDX
+    if [ "${KATA_HYPERVISOR}" = "qemu-tdx" ]; then
+	    skip "Test skipped on ${KATA_HYPERVISOR}, see https://github.com/kata-containers/kata-containers/issues/10011"
+    fi
+
+    if ! is_confidential_runtime_class; then
+        skip "Test not supported for ${KATA_HYPERVISOR}."
+    fi
+
+    [ "${SNAPSHOTTER:-}" = "nydus" ] || skip "None snapshotter was found but this test requires one"
```

Comment:
Should we also check that KBS is true because we rely on attestation here?

Reply:
We have:
```
    if [ "${KBS}" = "false" ]; then
        skip "Test skipped as KBS not setup"
    fi
```
in `setup_kbs_credentials` which is called in 2/3 tests, but the KBS isn't strictly needed in: the last one:
`"Test that creating a container from an authenticated image, with no credentials fails"`
Does that cover it for you?

Reply:
ah yeah that's fine